src/controller/CHIPCluster.h

```diff
@@ -186,16 +186,18 @@ class DLL_EXPORT ClusterBase
      * Read an attribute and get a type-safe callback with the attribute value.
      */
     template <typename AttributeInfo>
-    CHIP_ERROR ReadAttribute(void * context, ReadResponseSuccessCallback<typename AttributeInfo::DecodableArgType> successCb,
+    CHIP_ERROR ReadAttribute(void * context, const Optional<DataVersion> & aDataVersion,
```

Comment:
Why not put this arg last, defaulting to NullOptional, so only consumers who care about it will need to pass it in?  This applies to all the CHIPCluster APIs.

And then you won't need changes to chip-tool, CHIPDeviceController, etc.

Reply:
yes, agree, thanks